match-media.js

```diff
@@ -140,9 +143,14 @@
     // The 'scope' parameter is optional. If it's not passed in, '$rootScope' is used.
     this.when = function(list, callback, scope) {
       window.addEventListener('resize', function(event) {
-        if (that.is(list) === true) {
-          safeApply(callback(that.is(list)), scope);
-        }
+
+        // iOS fix: Has the window size really changed? (iOS sends resize event on scroll too)
+        if (window.innerWidth !== that.innerWidth) {
+          if (that.is(list) === true) {
+            safeApply(callback(that.is(list)), scope);
```

Comment:
Shouldn't you also re-set `that.innerWidth` here? As I read it, if somebody changes the window size back to the original, the callback won't get called..